cmd/ig/utils/flags.go

```diff
@@ -156,6 +160,13 @@ func AddCommonFlags(command *cobra.Command, commonFlags *CommonFlags) {
 		constants.K8sContainerdNamespace,
 		"Namespace used by containerd",
 	)
+
+	command.PersistentFlags().BoolVar(
+		&commonFlags.UseCri,
+		"use-cri",
+		false,
+		"Use CRI API to retrieve more K8s information, but ignore non K8s containers",
```

Comment:
@mauriciovasquezbernal Do you have an opinion on this flag?

In the case of containerd, `ig` connects to the same socket regardless of this flag but it would use a different gRPC protocol to get the information: CRI or internal containerd protocol.

In the case of docker, it is a different socket path for cri-dockerd than the main docker socket.

So the `--use-cri` interacts with the `--runtimes` flag:
```
  -r, --runtimes string                 Container runtimes to be used separated by comma. Supported values are: docker, containerd, cri-o, podman (default "docker,containerd,cri-o,podman")
```

`cri` cannot just be a possible choice in `--runtimes` because the runtime setting is used to know which socket to connect to. Unless we add `cri-docker`, `cri-containerd`, etc. but then it is more difficult to configure for the user.

I am not sure how to make it easier for the user, so I would keep the flag as it is for now, but it does not look ideal.

Reply:
pinging @mauriciovasquezbernal